src/app/public/modules/flyout/flyout-media-query.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Injectable
+} from '@angular/core';
+
+import {
+  BehaviorSubject
+} from 'rxjs/BehaviorSubject';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import {
+  SkyMediaBreakpoints
+} from '@skyux/core';
+
+import {
+  SkyMediaQueryListener
+} from '@skyux/core/modules/media-query/media-query-listener';
+
+import {
+  SkyFlyoutAdapterService
+} from './flyout-adapter.service';
+
+@Injectable()
+export class SkyFlyoutMediaQueryService {
+
+  public get current(): SkyMediaBreakpoints {
+    return this._current;
+  }
+
+  private currentSubject = new BehaviorSubject<SkyMediaBreakpoints>(this.current);
+
+  private _current = SkyMediaBreakpoints.lg;
```

Comment:
Why do we default to large?

Reply:
It is somewhat arbitrary. I wanted to make sure it was set to something on load. I went with large because that is the most common case.

Reply:
Most common? Do we have data on that?

Reply:
Nope :) That is just my opinion - that the majority of our users are not on small screens. Though, I guess this is for flyouts so maybe it should be defaulted to xs as most flyouts render at that size unless specified not to.

The main reason I did this was because I know there is code that we have which assumes that the media query service will return some breakpoint. Without a default that code will break. 

The flyout code resets the breakpoint to whatever is needed on initialization anyway just for full clarity. But I agree that we need to be deliberate here.

Reply:
In that case, I wonder if we should just go with `xs` to follow our philosophy of "mobile first". What do you think?

Reply:
I've updated it.

Reply:
perfect, thanks!